### DISCUSSION TITLE
Status of Foxy RADI 4.x

### DISCUSSION THREAD

**Discussion by pawanw17** [JdeRobot/RoboticsAcademy#1970] [Q&A]
Hi I am able to run docker images of other distros melodic and noetic well, However when trying to run the RADI of foxy it just shows up a terminal, I wanted to know what is the current status of foxy migration and whether I am running it correctly.

Command used:
`docker run --rm -it --device /dev/dri -p 8000:8000 -p 2303:2303 -p 1905:1905 -p 8765:8765 -p 6080:6080 -p 1108:1108 jderobot/robotics-academy:4.1.0 
`

![image](https://user-images.githubusercontent.com/40732709/225166181-bfe8d927-9274-4d56-8e6a-28167db28fd7.png)
@pariaspe @jmplaza

**Comment by jmplaza** [comment]
Hi @pawanw17,

RADI 4.3.X (with ROS2 Humble) is under development and @Blancasr has a working prototype. It will be the LTS RADI for a long time and all exercises and robot models will be eventually updated to work with ROS2 Humble. Here we are currently working with Gazebo 11, but it is planned the migration to Gazebo Fortress in the midterm.

RADI 4.2 with ROS2 Foxy is a draft, and it is not supported anymore. With it we developed the Follow-Person exercise, both with [simulated TurtleBot2 robot](https://www.youtube.com/watch?v=fDAU465eVxQ) and with [physical one](https://www.youtube.com/watch?v=54Jb4KJwyDM). Both will be migrated to RADI 4.3.X soon.

RADI 3.X (=with ROS1 Noetic) is currently the reference one for RoboticsAcademy. All available exercises are operational with it 

RADI with ROS1 Melodic is just outdated and deprecated.

**Reply by pawanw17** [reply]
Hi, Thanks for the update! I want to explore the RADI 4.2 more. If possible can you direct me where I can find the instructions to run it correctly, that way I can reverse engineer some things required for ROS 2. I believe migration from foxy to humble would be comparatively easier than noetic to humble

**Reply by pariaspe** [reply]
Hi Pawan, you can find them at foxy branch scripts [folder](https://github.com/JdeRobot/RoboticsAcademy/tree/foxy/scripts). Anyway, I'm not sure if it'll be helpful since we recently change the manager (the one in charge of launching the exercise).

**Reply by jmplaza** [reply]
Hi @pawanw17 , we have a brand new RADI 4.3.0 at https://github.com/JdeRobot/CustomRobots/tree/humble-devel
The RADI 4.3.0 (for dockerized execution of Robotics Applications) is built from that branch
It works, but it is the very first prototype. Feel free to ask any question at that repository creating an issue or on the Discussions of that repo. 
Hopefully next week we will publish the RADI 4.3.1 with the FollowLine exercise fully operative @Blancasr is working on it

**Reply by pawanw17** [reply]
Hi, Thanks for the update! I am able to run the 4.3.0 RADI.

 I had a doubt since this work is largely similar to project 2 of GSoC 23. Will the scope of the project change? @pariaspe can you share more on this, please?

**Reply by jmplaza** [reply]
That RADI 4.3.0 is the initial step, there is large room for contributions in the "migration of RoboticsAcademy to ROS2" global goal.  For instance migrating the support of ALL the robots used in the RADI 3.3.1 exercises (autonomous cars, drones, vacuum cleaners, cameras, industrial robots ...). Also taking benefit of Rviz2 in all the exercises for debugging. Also using the rosbag2 tools for some computer-vision exercises. 

In addition, we are working in robotics applications with several files (not only a single file as in RADI 3.3.1). Showing the computing graph in the browser for debugging purposes is a desired subgoal too. Also using some ROS2 stack (for instance rosnavigation2) for several exercises...

Moreover, we are working in supporting raw robotics applications (those without any template --all RADI 3.3.1 exercises use a Python template, for instance exercise.py--). In these applications the RoboticsAcademy user would have to explicitely program the ROS2 topics subscription, publication.... . This way, the creation of some pure ROS2 exercises, focused on learning how to use ROS2 topics or some ROS2 feature itself would be good contributions too.

**Reply by pawanw17** [reply]
@jmplaza Thanks for the suggestions! I'll make sure to include them in the proposal.

**Reply by pawanw17** [reply]
@jmplaza by robot models do you mean the URDFs? if yes are URDFs updated with ROS2?

**Reply by jmplaza** [reply]
yes, URDFs, SDFs and plugins...
